node/heartbeat/monitor.go

```diff
@@ -88,7 +88,14 @@ func (m *Monitor) ProcessReceivedMessage(message p2p.MessageP2P) error {
 		return err
 	}
 
-	err = m.singleSigner.Verify(senderPubkey, hbRecv.Payload, hbRecv.Signature)
+	copiedHeartbeat := *hbRecv
```

Comment:
move signature verification to a separate function.

Reply:
done